File: Tarea 1/zlatan.py
"""
    Arboles y Grafos 2025-1
    Santiago Guevara Idarraga
    Problem E - zlatan
    Enero 23 de 2025
"""

from heapq import heappush, heappop

class Card:
    def __init__(self, lbl, srl):
        self.label = lbl
        self.serial = srl
        self.ocurrences = 1
def main():
    n, m = map(int, input().split())
    while n != 0 and m != 0:
        dic, pq = {}, []
        while n != 0:
            label, serial = input().split()
            if label in dic:
                dic[label].ocurrences += 1
            else:
                dic[label] = Card(label, serial)
            n -= 1
        for key in dic:
            heappush(pq, dic[key])
        while m != 0:
            print(f"{pq[0].label} {pq[0].ocurrences}")
            m -= 1
        n, m = map(int, input().split())

"""
def main():
    n, m = map(int, input().split())
    while n != 0 and m != 0:
        dic, pq = {}, []
        for _ in range(n):
            label, serial = input().split()
            key = (label, serial)
            if key in dic:
                dic[key] += 1
            else:
                dic[key] = 1
        for key in dic:
            heappush(pq, (-dic[key], (key[0], key[1])))
        for _ in range(m):
            counter, (label, serial) = heappop(pq)
            print(f"{label} {-counter}")
        n, m = map(int, input().split())
"""

main()

File: Tarea 1/zlatan.cpp
/*
    Arboles y Grafos 2025-1
    Santiago Guevara Idarraga
    Problem E - zlatan
    Enero 29 de 2025
*/

#include <iostream>
#include <map>
#include <queue>
#include <vector>
#include <algorithm>

using namespace std;

struct Card {
    string label;
    int serial, ocurrences;
    Card(string lbl, int srl) {
        label = lbl;
        serial = srl;
        ocurrences = 1;
    }
    bool operator<(const Card &b) const {
        bool ans = false;
        if(ocurrences != b.ocurrences) {
            ans = ocurrences < b.ocurrences;
        }
        else if(serial != b.serial) {
            ans = serial < b.serial;
        }
        return ans;
    }
};
int main() {
    int n, m, serial;
    string label;
    map<string, Card> data;
    vector<Card> v;
    map<string, Card>::iterator it;
    while(cin >> n >> m && n != 0 && m != 0) {
        priority_queue<Card> pq;
        while(n--) {
            cin >> label >> serial;
            it = data.find(label);
            if(it != data.end()) {
                it->second.ocurrences++;
            }
            else {
                data.insert({label, Card(label, serial)});
            }
        }
        for(it = data.begin(); it != data.end(); it++) {
            pq.push(it->second);
        }
        for(int i = 0; i < m; i++) {
            v.push_back(pq.top());
            pq.pop();
        }
        sort(v.begin(), v.end(), [](const Card &a, const Card &b){return a.label < b.label;});
        for(vector<Card>::iterator jt = v.begin(); jt != v.end(); jt++) {
            cout << jt->label << ' ' << jt->ocurrences << endl;
        }
        data.clear();
        v.clear();
    }
    return 0;
}